pkg/virt-controller/watch/vm_test.go

```diff
@@ -5161,6 +5161,65 @@ var _ = Describe("VirtualMachine", func() {
 					vmi := controller.setupVMIFromVM(vm)
 					Expect(vmi.Spec.Domain.CPU.MaxSockets).To(Equal(defaultSockets * 4))
 				})
+
+				DescribeTable("should patch VMI when CPU hotplug is requested", func(resources virtv1.ResourceRequirements) {
+					vm, _ := DefaultVirtualMachine(true)
+					vm.Spec.Template.Spec.Domain.Resources = resources
+					vm.Spec.Template.Spec.Domain.CPU = &virtv1.CPU{
+						Sockets: 2,
+					}
+
+					vmi := api.NewMinimalVMI(vm.Name)
+					vmi.Spec.Domain.CPU = &virtv1.CPU{
+						Sockets:    1,
+						MaxSockets: 4,
+					}
+					vmi.Spec.Domain.Resources = resources
+
+					vmiInterface.EXPECT().Patch(context.Background(), vmi.Name, types.JSONPatchType, gomock.Any(), metav1.PatchOptions{}).Do(
+						func(ctx context.Context, name, patchType, patch, opts interface{}, subs ...interface{},
```

Comment:
@acardace I feel less confident with the newline here, but if it complies to some standard, I am fine.
Can you please elaborate on the `opts interface{}, subs ...interface{}` args? why are they needed?

Reply:
About the args, that's just how it is declared by the mock, https://github.com/kubevirt/kubevirt/blob/main/staging/src/kubevirt.io/client-go/kubecli/generated_mock_kubevirt.go#L1128.

The test would otherwise fail to run as the passed function would have the wrong number of arguments.

Reply:
About the newline, I removed it, I think it was a mistake.

Reply:
@enp0s3 PTAL